src/containers/post/PostProvider.js

```diff
@@ -0,0 +1,91 @@
+import {
+  createContext,
+  useContext,
+  useState,
+  useMemo,
+  useCallback,
+} from "react";
+import PropTypes from "prop-types";
+
+import * as postAPIs from "../../apis/posts";
+import { showSuccessAlert } from "../../store/directDispatches/common";
+
+const PostContext = createContext({});
+
+function PostProvider(props) {
+  const { children } = props;
+  const [postData, setPostData] = useState([]);
+
+  const getPostData = useCallback(async () => {
+    try {
+      console.log("Initiating retrieving admin posts");
+      const { success, data } = await postAPIs.getPosts();
+      if (success) {
+        setPostData(data);
+        console.log("Successfully retrieved admin posts", data);
+      }
+    } catch (e) {
+      console.log("Unable to retrieve admin posts", e);
+    }
+  }, [setPostData]);
```

Comment:
do we need to add setPostData function as a dependancy?

Reply:
Fixed